cirq/google/engine/client/quantum_v1alpha1/gapic/quantum_engine_service_client.py

```diff
@@ -74,13 +76,16 @@ def from_service_account_file(cls, filename, *args, **kwargs):
 
     from_service_account_json = from_service_account_file
 
-    def __init__(self,
-                 transport=None,
-                 channel=None,
-                 credentials=None,
-                 client_config=None,
-                 client_info=None,
-                 client_options=None):
+    def __init__(
+            self,
+            transport: Union[quantum_engine_service_grpc_transport.
+                             QuantumEngineServiceGrpcTransport] = None,
+            channel: grpc.Channel = None,
```

Comment:
Since you're allowing `None` this should be `Optional[grpc.Channel]`. This applies throughout the PR.

Why hasn't mypy caught any of those?

Reply:
Wrapped with `Optional`.

The note just above [this header](https://mypy.readthedocs.io/en/stable/kinds_of_types.html#disabling-strict-optional-checking) in the mypy docs explains this: arguments with a default value of `None` aren't required to be `Optional` unless `--no-implicit-optional` is enabled.